### ISSUE TITLE
Cannot yield list of coroutines and triggers?

### ISSUE THREAD

**Issue by TC01** [potentialventures/cocotb#588]
Is it intentional that I can only yield a list of triggers, not a list of triggers *and* coroutines? According to [this code in the scheduler](https://github.com/potentialventures/cocotb/blob/master/cocotb/scheduler.py#L642), yielding a list works only when the list contains nothing but triggers.

```
        elif (isinstance(result, list) and
                not [t for t in result if not isinstance(t, Trigger)]):
            self._coroutine_yielded(coroutine, result)
```

What I'd like to do here is have a timeout trigger that fires and causes the simulation to stop waiting for a coroutine after a certain point. Unfortunately it seems like that's not possible.

**Comment by forensicgarlic** [comment]
I've done stuff like 
```python
    t = Timer(int(dut.PERIOD) * CLK_PERIOD * 2)
    result = yield [RisingEdge(dut.pulse_out), t]
    if result == t:
        raise TestFailure("output is not toggling")
```
successfully in the past. Perhaps your coroutine is acting on a signal that you could detect an edge on as well?

**Comment by TC01** [comment]
Oh, that's clever, forking and using the trigger generated by ```coroutine.join()```. I suppose that should work!

I think it would be nicer if cocotb could do this automatically, though. Presumably it's a pretty simple modification to the code in the scheduler... perhaps I'll submit a patch.

**Comment by TC01** [comment]
I finally got around to looking at this. I wrote a pretty simple patch for the scheduler to make it capable of handling coroutines in a list of triggers, and wrote a small test case. See #618.

**Comment by eric-wieser** [comment]
Closed by #618 I think

**Comment by eric-wieser** [comment]
Ping @imphil, close?
